REST Data Structure Controller

Sopczasty opened the pull request:
Składniki na 3 makowce:

Ciasto drożdżowe:

500 g mąki pszennej + do podsypywania
180 ml mleka
150 g roztopionego i ostudzonego masła (lub margaryny)
6 żółtek
42 g świeżych drożdży (lub 21 g suchych)
6 łyżek cukru
½ łyżeczki soli
1,5 łyżki spirytusu (lub rumu, wódki)
2 łyżeczki cukru waniliowego

Masa makowa:
500 g maku
200 g cukru
100 g rodzynek
50 g posiekanych orzechów włoskich
1 łyżka miodu
aromat migdałowy do smaku
szczypta cynamonu
1 łyżka miękkiego masła
ewentualnie kandyzowana skórka pomarańczowa
6 białek

Lukier:
1 szklanka cukru pudru
3-4 łyżki gorącej wody
Sposób przygotowania:

1. Mak zalać wrzątkiem do przykrycia, przykryć talerzykiem i pozostawić na co najmniej 30 minut. Odkryć i pozostawić do całkowitego ostygnięcia. Mak odcedzić na sitku wyłożonym gazą lub ściereczką i dobrze odcisnąć. Jeśli nie jest zmielony należy go zmielić dwukrotnie w maszynce do mielenia mięsa, używając sitka o drobnych oczkach. (Zmielony mak jest już gotowy po odciśnięciu do dalszego przygotowania).

2. Lekko ciepłe mleko wymieszać z 1 łyżeczką cukru, pokruszonymi drożdżami i ok. 4 łyżkami mąki. Przykryć ściereczką i odstawić w ciepłe miejsce do wyrośnięcia na ok. 10- 15 minut.

3. Wyrośnięty zaczyn połączyć z pozostałymi składnikami na ciasto drożdżowe (poza masłem). Zagniatać ciasto, wlewając stopniowo roztopione masło (lub margarynę). Przykryć ściereczką i pozostawić do wyrośnięcia, aż podwoi swoją objętość (ok. 1- 1,5 godziny).

4. Do maku dodać pozostałe składniki masy, poza białkami. Białka ubić na sztywną pianę. Dodać do masy i delikatnie wymieszać szpatułką lub łyżką drewnianą.

5. Ciasto podzielić na 3 części. Każdą część rozwałkować na prostokąt o grubości ok. 3 mm, nałożyć ⅓ masy, zostawiając ok. 2 cm od brzegu i zwinąć z roladę. Końce rolady skleić i podwinąć pod spód. Gotową roladę przełożyć na posmarowany olejem papier do pieczenia i zrolować w formie tuby dwukrotnie, pozostawiając 1 cm luzu między ciastem, a papierem (nie więcej). Końce papieru pozostawić otwarte. Makowce przenieść na blachę do pieczenia.

6. Piec w nagrzanym piekarniku ok. 40 minut w temperaturze 190°C (grzałka góra- dół), na brązowo- złoty kolor. Pozostawić do ostygnięcia.

7. Jeszcze ciepłe makowce (nie gorące) posmarować lukrem.

8. Przygotować lukier: Cukier puder rozetrzeć z gorącą wodą na gładką masę. Makowce polać lukrem i ewentualnie posypać makiem. (Chyba, że chcemy makowce zamrozić, wtedy lukier zrobić po rozmrożeniu ciasta).

Uwagi:
Dłuższy bok rozwałkowanego prostokąta powinien być krótszy niż długość blachy tak, aby makowiec spokojnie zmieścił się w piekarniku, biorąc zapas, że jeszcze na boki urośnie. Ja rozwałkowałam ciasto na prostokąt 32x 25 cm.

Nie polecam używania gotowej masy makowej, ponieważ jest za słodka, z niezdrowymi i niepotrzebnymi dodatkami, często za rzadka i wpływa na jakość makowca. Jeśli chcemy skrócić czas przygotowania to polecam kupno gotowego, zmielonego maku. Jeśli zdecydujecie się jednak na gotową masę makową z puszki to według sugestii z komentarzy zaleca się dodanie do niej dwóch ubitych na sztywno białek. Ilość gotowej masy z puszki na trzy makowce to ok. 1,5 dużej puszki (1 puszka zawiera ok. 850g).

Sopczasty committed:
Składniki na 3 makowce:
Ciasto drożdżowe:

500 g mąki pszennej + do podsypywania
180 ml mleka
150 g roztopionego i ostudzonego masła (lub margaryny)
6 żółtek
42 g świeżych drożdży (lub 21 g suchych)
6 łyżek cukru
½ łyżeczki soli
1,5 łyżki spirytusu (lub rumu, wódki)
2 łyżeczki cukru waniliowego

Masa makowa:
500 g maku
200 g cukru
100 g rodzynek
50 g posiekanych orzechów włoskich
1 łyżka miodu
aromat migdałowy do smaku
szczypta cynamonu
1 łyżka miękkiego masła
ewentualnie kandyzowana skórka pomarańczowa
6 białek

Lukier:
1 szklanka cukru pudru
3-4 łyżki gorącej wody
Sposób przygotowania:

1. Mak zalać wrzątkiem do przykrycia, przykryć talerzykiem i pozostawić na co najmniej 30 minut. Odkryć i pozostawić do całkowitego ostygnięcia. Mak odcedzić na sitku wyłożonym gazą lub ściereczką i dobrze odcisnąć. Jeśli nie jest zmielony należy go zmielić dwukrotnie w maszynce do mielenia mięsa, używając sitka o drobnych oczkach. (Zmielony mak jest już gotowy po odciśnięciu do dalszego przygotowania).

2. Lekko ciepłe mleko wymieszać z 1 łyżeczką cukru, pokruszonymi drożdżami i ok. 4 łyżkami mąki. Przykryć ściereczką i odstawić w ciepłe miejsce do wyrośnięcia na ok. 10- 15 minut.

3. Wyrośnięty zaczyn połączyć z pozostałymi składnikami na ciasto drożdżowe (poza masłem). Zagniatać ciasto, wlewając stopniowo roztopione masło (lub margarynę). Przykryć ściereczką i pozostawić do wyrośnięcia, aż podwoi swoją objętość (ok. 1- 1,5 godziny).

4. Do maku dodać pozostałe składniki masy, poza białkami. Białka ubić na sztywną pianę. Dodać do masy i delikatnie wymieszać szpatułką lub łyżką drewnianą.

5. Ciasto podzielić na 3 części. Każdą część rozwałkować na prostokąt o grubości ok. 3 mm, nałożyć ⅓ masy, zostawiając ok. 2 cm od brzegu i zwinąć z roladę. Końce rolady skleić i podwinąć pod spód. Gotową roladę przełożyć na posmarowany olejem papier do pieczenia i zrolować w formie tuby dwukrotnie, pozostawiając 1 cm luzu między ciastem, a papierem (nie więcej). Końce papieru pozostawić otwarte. Makowce przenieść na blachę do pieczenia.

6. Piec w nagrzanym piekarniku ok. 40 minut w temperaturze 190°C (grzałka góra- dół), na brązowo- złoty kolor. Pozostawić do ostygnięcia.

7. Jeszcze ciepłe makowce (nie gorące) posmarować lukrem.

8. Przygotować lukier: Cukier puder rozetrzeć z gorącą wodą na gładką masę. Makowce polać lukrem i ewentualnie posypać makiem. (Chyba, że chcemy makowce zamrozić, wtedy lukier zrobić po rozmrożeniu ciasta).

Uwagi:
Dłuższy bok rozwałkowanego prostokąta powinien być krótszy niż długość blachy tak, aby makowiec spokojnie zmieścił się w piekarniku, biorąc zapas, że jeszcze na boki urośnie. Ja rozwałkowałam ciasto na prostokąt 32x 25 cm.

Nie polecam używania gotowej masy makowej, ponieważ jest za słodka, z niezdrowymi i niepotrzebnymi dodatkami, często za rzadka i wpływa na jakość makowca. Jeśli chcemy skrócić czas przygotowania to polecam kupno gotowego, zmielonego maku. Jeśli zdecydujecie się jednak na gotową masę makową z puszki to według sugestii z komentarzy zaleca się dodanie do niej dwóch ubitych na sztywno białek. Ilość gotowej masy z puszki na trzy makowce to ok. 1,5 dużej puszki (1 puszka zawiera ok. 850g).

Sopczasty committed:
Składniki na 3 makowce:

Ciasto drożdżowe:

500 g mąki pszennej + do podsypywania
180 ml mleka
150 g roztopionego i ostudzonego masła (lub margaryny)
6 żółtek
42 g świeżych drożdży (lub 21 g suchych)
6 łyżek cukru
½ łyżeczki soli
1,5 łyżki spirytusu (lub rumu, wódki)
2 łyżeczki cukru waniliowego

Masa makowa:
500 g maku
200 g cukru
100 g rodzynek
50 g posiekanych orzechów włoskich
1 łyżka miodu
aromat migdałowy do smaku
szczypta cynamonu
1 łyżka miękkiego masła
ewentualnie kandyzowana skórka pomarańczowa
6 białek

Lukier:
1 szklanka cukru pudru
3-4 łyżki gorącej wody
Sposób przygotowania:

Mak zalać wrzątkiem do przykrycia, przykryć talerzykiem i pozostawić na co najmniej 30 minut. Odkryć i pozostawić do całkowitego ostygnięcia. Mak odcedzić na sitku wyłożonym gazą lub ściereczką i dobrze odcisnąć. Jeśli nie jest zmielony należy go zmielić dwukrotnie w maszynce do mielenia mięsa, używając sitka o drobnych oczkach. (Zmielony mak jest już gotowy po odciśnięciu do dalszego przygotowania).

Lekko ciepłe mleko wymieszać z 1 łyżeczką cukru, pokruszonymi drożdżami i ok. 4 łyżkami mąki. Przykryć ściereczką i odstawić w ciepłe miejsce do wyrośnięcia na ok. 10- 15 minut.

Wyrośnięty zaczyn połączyć z pozostałymi składnikami na ciasto drożdżowe (poza masłem). Zagniatać ciasto, wlewając stopniowo roztopione masło (lub margarynę). Przykryć ściereczką i pozostawić do wyrośnięcia, aż podwoi swoją objętość (ok. 1- 1,5 godziny).

Do maku dodać pozostałe składniki masy, poza białkami. Białka ubić na sztywną pianę. Dodać do masy i delikatnie wymieszać szpatułką lub łyżką drewnianą.

Ciasto podzielić na 3 części. Każdą część rozwałkować na prostokąt o grubości ok. 3 mm, nałożyć ⅓ masy, zostawiając ok. 2 cm od brzegu i zwinąć z roladę. Końce rolady skleić i podwinąć pod spód. Gotową roladę przełożyć na posmarowany olejem papier do pieczenia i zrolować w formie tuby dwukrotnie, pozostawiając 1 cm luzu między ciastem, a papierem (nie więcej). Końce papieru pozostawić otwarte. Makowce przenieść na blachę do pieczenia.

Piec w nagrzanym piekarniku ok. 40 minut w temperaturze 190°C (grzałka góra- dół), na brązowo- złoty kolor. Pozostawić do ostygnięcia.

Jeszcze ciepłe makowce (nie gorące) posmarować lukrem.

Przygotować lukier: Cukier puder rozetrzeć z gorącą wodą na gładką masę. Makowce polać lukrem i ewentualnie posypać makiem. (Chyba, że chcemy makowce zamrozić, wtedy lukier zrobić po rozmrożeniu ciasta).

Uwagi:
Dłuższy bok rozwałkowanego prostokąta powinien być krótszy niż długość blachy tak, aby makowiec spokojnie zmieścił się w piekarniku, biorąc zapas, że jeszcze na boki urośnie. Ja rozwałkowałam ciasto na prostokąt 32x 25 cm.

Nie polecam używania gotowej masy makowej, ponieważ jest za słodka, z niezdrowymi i niepotrzebnymi dodatkami, często za rzadka i wpływa na jakość makowca. Jeśli chcemy skrócić czas przygotowania to polecam kupno gotowego, zmielonego maku. Jeśli zdecydujecie się jednak na gotową masę makową z puszki to według sugestii z komentarzy zaleca się dodanie do niej dwóch ubitych na sztywno białek. Ilość gotowej masy z puszki na trzy makowce to ok. 1,5 dużej puszki (1 puszka zawiera ok. 850g).

Dawnkai commented on src/test/java/pl/put/poznan/sorting/rest/DataStructureControllerTest.java:
```diff
@@ -0,0 +1,119 @@
+        assertTrue(response.hasBody());
+        assertTrue(response.getBody().get("error").toString().contains("Wrong payload content"));
+    }
+}
```
`no newline at the end of file`

Dawnkai commented on src/test/java/pl/put/poznan/sorting/rest/ArraySorterControllerTest.java:
```diff
@@ -1,113 +1,117 @@
+        System.out.println(response.getBody().get("result").toString());
+        for(int i = 0; i < expected.length; i++)
+            array[i] = expected[i].toString();
+        assertArrayEquals(array, response.getBody().get("result").toString()
```
`replaceAll` zamiast `replace`

Dawnkai commented on src/main/java/pl/put/poznan/sorting/rest/ArraySorterController.java:
```diff
@@ -19,64 +20,81 @@
         HashMap<String, Object> output = new HashMap<String, Object>();
         output.put("result", result);
         output.put("algorithm", algorithm);
+        output.put("elapsed", madness.getTimer().getTimeElapsed());
```
Od tego są statystyki, a nie Timer.

Dawnkai commented on src/main/java/pl/put/poznan/sorting/rest/ArraySorterController.java:
```diff
@@ -19,64 +20,81 @@
+     * @param data; String of data delimited by ','
+     * @return Object[] array of the given type
+     */
+    private Object[] detectType(String data) {
```
Nie duplikuj kodu z `App`. Jeżeli tego potrzebujesz, przenieś tę funkcję do `SortingMadness.java` (skoro i tak importujesz stamtąd klasę) - i ją wywołaj i tu i w `App`.

Sopczasty committed:
Składniki na 3 makowce:

Ciasto drożdżowe:

500 g mąki pszennej + do podsypywania
180 ml mleka
150 g roztopionego i ostudzonego masła (lub margaryny)
6 żółtek
42 g świeżych drożdży (lub 21 g suchych)
6 łyżek cukru
½ łyżeczki soli
1,5 łyżki spirytusu (lub rumu, wódki)
2 łyżeczki cukru waniliowego

Masa makowa:
500 g maku
200 g cukru
100 g rodzynek
50 g posiekanych orzechów włoskich
1 łyżka miodu
aromat migdałowy do smaku
szczypta cynamonu
1 łyżka miękkiego masła
ewentualnie kandyzowana skórka pomarańczowa
6 białek

Lukier:
1 szklanka cukru pudru
3-4 łyżki gorącej wody
Sposób przygotowania:

Mak zalać wrzątkiem do przykrycia, przykryć talerzykiem i pozostawić na co najmniej 30 minut. Odkryć i pozostawić do całkowitego ostygnięcia. Mak odcedzić na sitku wyłożonym gazą lub ściereczką i dobrze odcisnąć. Jeśli nie jest zmielony należy go zmielić dwukrotnie w maszynce do mielenia mięsa, używając sitka o drobnych oczkach. (Zmielony mak jest już gotowy po odciśnięciu do dalszego przygotowania).

Lekko ciepłe mleko wymieszać z 1 łyżeczką cukru, pokruszonymi drożdżami i ok. 4 łyżkami mąki. Przykryć ściereczką i odstawić w ciepłe miejsce do wyrośnięcia na ok. 10- 15 minut.

Wyrośnięty zaczyn połączyć z pozostałymi składnikami na ciasto drożdżowe (poza masłem). Zagniatać ciasto, wlewając stopniowo roztopione masło (lub margarynę). Przykryć ściereczką i pozostawić do wyrośnięcia, aż podwoi swoją objętość (ok. 1- 1,5 godziny).

Do maku dodać pozostałe składniki masy, poza białkami. Białka ubić na sztywną pianę. Dodać do masy i delikatnie wymieszać szpatułką lub łyżką drewnianą.

Ciasto podzielić na 3 części. Każdą część rozwałkować na prostokąt o grubości ok. 3 mm, nałożyć ⅓ masy, zostawiając ok. 2 cm od brzegu i zwinąć z roladę. Końce rolady skleić i podwinąć pod spód. Gotową roladę przełożyć na posmarowany olejem papier do pieczenia i zrolować w formie tuby dwukrotnie, pozostawiając 1 cm luzu między ciastem, a papierem (nie więcej). Końce papieru pozostawić otwarte. Makowce przenieść na blachę do pieczenia.

Piec w nagrzanym piekarniku ok. 40 minut w temperaturze 190°C (grzałka góra- dół), na brązowo- złoty kolor. Pozostawić do ostygnięcia.

Jeszcze ciepłe makowce (nie gorące) posmarować lukrem.

Przygotować lukier: Cukier puder rozetrzeć z gorącą wodą na gładką masę. Makowce polać lukrem i ewentualnie posypać makiem. (Chyba, że chcemy makowce zamrozić, wtedy lukier zrobić po rozmrożeniu ciasta).

Uwagi:
Dłuższy bok rozwałkowanego prostokąta powinien być krótszy niż długość blachy tak, aby makowiec spokojnie zmieścił się w piekarniku, biorąc zapas, że jeszcze na boki urośnie. Ja rozwałkowałam ciasto na prostokąt 32x 25 cm.

Nie polecam używania gotowej masy makowej, ponieważ jest za słodka, z niezdrowymi i niepotrzebnymi dodatkami, często za rzadka i wpływa na jakość makowca. Jeśli chcemy skrócić czas przygotowania to polecam kupno gotowego, zmielonego maku. Jeśli zdecydujecie się jednak na gotową masę makową z puszki to według sugestii z komentarzy zaleca się dodanie do niej dwóch ubitych na sztywno białek. Ilość gotowej masy z puszki na trzy makowce to ok. 1,5 dużej puszki (1 puszka zawiera ok. 850g).

Dawnkai committed:
Fix documentation errors in SortingMadness class

Dawnkai committed:
Fix documentation error in REST

Dawnkai committed:
Bump version
